app/scripts/metamask-controller.js

```diff
@@ -4346,117 +4343,59 @@ export default class MetamaskController extends EventEmitter {
   // Log blocks
 
   /**
-   * A method for selecting a custom URL for an ethereum RPC provider and updating it
+   * Adds a network configuration if the rpcUrl is not already present on an
+   * existing network configuration. Otherwise updates the entry with the matching rpcUrl.
    *
-   * @param {string} rpcUrl - A URL for a valid Ethereum RPC API.
-   * @param {string} chainId - The chainId of the selected network.
-   * @param {string} ticker - The ticker symbol of the selected network.
-   * @param {string} [nickname] - Nickname of the selected network.
-   * @param {object} [rpcPrefs] - RPC preferences.
-   * @param {string} [rpcPrefs.blockExplorerUrl] - URL of block explorer for the chain.
-   * @returns {Promise<string>} The RPC Target URL confirmed.
+   * @param config - The network configuration.
+   * @param config.rpcUrl - The network configuration RPC URL.
+   * @param config.chainId - The chain ID of the network, as per EIP-155.
+   * @param config.ticker - Currency ticker.
+   * @param config.chainName - Personalized network name.
+   * @param config.rpcPrefs - Personalized preferences.
+   * @returns uuid for the added or updated network configuration
    */
-  async updateAndSetCustomRpc(
+  async upsertAndSetNetworkConfiguration({
```

Comment:
If we want to be literal, then I wonder if this should be called `upsertNetworkConfigurationAndSetNetworkTarget`. However, I feel like this is a mouthful to say. I wonder if it's worth naming this from a user's perspective, so something like `upsertAndConnectToNetwork`? (Also see the comment I left in NetworkController about the naming of `setNetworkTarget.)

Reply:
Perhaps `upsertAndSetActiveNetwork`?